homeassistant/components/homematic/__init__.py

```diff
@@ -740,6 +740,11 @@ def async_added_to_hass(self):
         """Load data init callbacks."""
         yield from self.hass.async_add_job(self.link_homematic)
 
+    @property
+    def unique_id(self):
+        """Return unique ID. HomeMatic entity IDs are unique by default."""
+        return self._name
```

Comment:
Please clarify if the user can make this non unique or if this will always be constant (unique). Reading this PR description and the linked issue, it's not clear to me what you mean might change due to setting friendly name (using the resolve names function).

Reply:
As long as the unique_id is unique and doesn't change for an entity, the entity_id for that entity will not change during restarts, even if the another entity_id is suggested (eg by the user).

That's the whole point of the entity registry. The entity registry is the truth for the entity_ids. So after an entity has been registered, the entity_id can only be changed via the entity registry or an api that uses the entity registry.